src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts

```diff
@@ -1,70 +1,77 @@
+import { Actions, ofType } from '@ngrx/effects';
 import { Store } from '@ngrx/store';
 import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
 import { TestRecordsService } from '@services/test-records/test-records.service';
-import { createProvisionalTechRecord, updateTechRecords } from '@store/technical-records';
+import { editingTechRecord, updateTechRecord, updateTechRecordSuccess } from '@store/technical-records';
 import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
-import { Observable, tap } from 'rxjs';
+import { Observable, Subject, take, takeUntil } from 'rxjs';
 import { TechRecordSummaryComponent } from '../tech-record-summary/tech-record-summary.component';
 
 @Component({
-  selector: 'app-vehicle-technical-record[vehicle]',
+  selector: 'app-vehicle-technical-record',
   templateUrl: './vehicle-technical-record.component.html',
   styleUrls: ['./vehicle-technical-record.component.scss']
 })
-export class VehicleTechnicalRecordComponent implements OnInit {
+export class VehicleTechnicalRecordComponent implements OnInit, OnDestroy {
   @ViewChild(TechRecordSummaryComponent) summary!: TechRecordSummaryComponent;
-  @Input() vehicle!: VehicleTechRecordModel;
+  @Input() techRecord?: V3TechRecordModel;
 
-  currentTechRecord$: Observable<TechRecordModel | undefined>;
   testResults$: Observable<TestResultModel[]>;
   editingReason?: ReasonForEditing;
+  recordHistory?: V3TechRecordModel[];
 
   isCurrent = false;
   isArchived = false;
   isEditing = false;
   isDirty = false;
   isInvalid = false;
 
+  destroy$ = new Subject();
+
   constructor(
     testRecordService: TestRecordsService,
     private activatedRoute: ActivatedRoute,
     private route: ActivatedRoute,
     private router: Router,
     private store: Store<TechnicalRecordServiceState>,
-    private technicalRecordService: TechnicalRecordService
+    private technicalRecordService: TechnicalRecordService,
+    private actions$: Actions
   ) {
     this.testResults$ = testRecordService.testRecords$;
     this.isEditing = this.activatedRoute.snapshot.data['isEditing'] ?? false;
     this.editingReason = this.activatedRoute.snapshot.data['reason'];
-    this.currentTechRecord$ = this.technicalRecordService.viewableTechRecord$.pipe(
-      tap(viewableTechRecord => {
-        this.isCurrent = viewableTechRecord?.statusCode === StatusCodes.CURRENT;
-        this.isArchived = viewableTechRecord?.statusCode === StatusCodes.ARCHIVED;
-      })
-    );
   }
-
+  ngOnDestroy(): void {
+    this.destroy$.next;
+    this.destroy$.complete();
+  }
   ngOnInit(): void {
-    const hasProvisionalRecord = this.vehicle.techRecord.some(record => record.statusCode === StatusCodes.PROVISIONAL);
+    const hasProvisionalRecord = this.techRecord?.techRecord_statusCode === StatusCodes.PROVISIONAL;
```

Comment:
isn't the logic here that we want to know if the vehicle has a provisional record? I think you'd have to get the techRecordHistory and check if there is a provisonal record